src/Unity.Mathematics/random.cs

```diff
@@ -27,6 +27,45 @@ public Random(uint seed)
             NextState();
         }
 
+        /// <summary>
+        /// Constructs a <see cref="Random"/> instance with a hashed seed.
+        /// </summary>
+        /// <remarks>
+        /// Use this function when you expect to create several Random instances from consecutive or
+        /// very similar seeds.
+        /// </remarks>
+        /// <example>
+        /// <code>
+        /// for (uint i = 0; i &lt; 4096; ++i)
+        /// {
+        ///     Random rand = Random.CreateFromHashedSeed(i);
+        ///
+        ///     // Random numbers drawn from loop iteration j should be very different
+        ///     // from every other loop iteration.
+        ///     rand.NextUInt();
+        /// }
+        /// </code>
+        /// </example>
+        /// <param name="seed_to_hash">A seed that will be hashed for Random creation.</param>
+        /// <param name="backup_nonzero_seed">In the event that <paramref name="seed_to_hash"/> hashes to zero, then the <see cref="Random"/> constructor will be called with this backup value as the seed instead.  No hashing is performed on the backup and it must be non-zero.</param>
+        /// <returns><see cref="Random"/> created from hashed seed.  If the seed hashes to zero, then the backup seed is used without hashing it.</returns>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static Random CreateFromHashedSeed(uint seed_to_hash, uint backup_nonzero_seed = 1u)
```

Comment:
The main thing I don't like about this function is that it's not clear how to choose the value for `backup_nonzero_seed` if you don't know how the hash behaves.  If it's critical that I do not have any collisions, then I need to dig into the implementation and see which hashes I generate for a given set of seeds, then pick some other seed that doesn't match any of the hashes.

Reply:
Would just reversing the bits of the seed prevent the correlation between consecutive seeds? That has the nice property that 0 is still the "bad" seed.

Alternatively, subtract off 61 before applying the Wang hash, so that 0 -> 0

Reply:
I tried reversing the bits but it doesn't do enough to pass the statistical tests.

> Alternatively, subtract off 61 before applying the Wang hash, so that 0 -> 0

Now *that's* the kind of thinking I've been failing to do!  I'll try that out.